txn.go

```diff
@@ -167,41 +171,85 @@ func (txn *Txn) Insert(table string, obj interface{}) error {
 
 		// Handle the update by deleting from the index first
 		if update {
-			ok, val, err := indexSchema.Indexer.FromObject(existing)
+			switch indexer := indexSchema.Indexer.(type) {
+			case SingleIndexer:
+				ok, val, err := indexer.FromObject(existing)
```

Comment:
May be cleaner to move `ok, vals, err` out of the switch and share the logic between both cases. E.g. the only thing inside the switch is the assign to the variables.